x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elastic/ElasticInferenceService.java

```diff
@@ -126,17 +164,63 @@ private void getAuthorization() {
         }
     }
 
-    private synchronized void setEnabledTaskTypes(ElasticInferenceServiceAuthorization auth) {
-        enabledTaskTypesRef.set(filterTaskTypesByAuthorization(auth));
-        configuration = new Configuration(enabledTaskTypesRef.get());
+    private synchronized void setAuthorizedContent(ElasticInferenceServiceAuthorization auth) {
+        var authorizedTaskTypesAndModels = auth.newLimitedToTaskTypes(EnumSet.copyOf(IMPLEMENTED_TASK_TYPES));
+
+        // recalculate which default config ids and models are authorized now
+        var enabledDefaultConfigIds = getEnabledDefaultConfigIds(auth);
+        var enabledDefaultModelObjects = getEnabledDefaultModelsObjects(auth);
+        authRef.set(new AuthorizedContent(authorizedTaskTypesAndModels, enabledDefaultConfigIds, enabledDefaultModelObjects));
+
+        configuration = new Configuration(authRef.get().authorizedTaskTypesAndModels.getEnabledTaskTypes());
 
         defaultConfigIds().forEach(modelRegistry::addDefaultIds);
     }
 
-    private static EnumSet<TaskType> filterTaskTypesByAuthorization(ElasticInferenceServiceAuthorization auth) {
-        var implementedTaskTypes = EnumSet.copyOf(IMPLEMENTED_TASK_TYPES);
-        implementedTaskTypes.retainAll(auth.enabledTaskTypes());
-        return implementedTaskTypes;
+    private List<DefaultConfigId> getEnabledDefaultConfigIds(ElasticInferenceServiceAuthorization auth) {
+        var enabledDefaultModelIds = getEnabledDefaultModelIds(auth);
+
+        var enabledConfigIds = new ArrayList<DefaultConfigId>();
+        for (var id : enabledDefaultModelIds) {
+            var model = defaultModels.get(id);
+            if (model != null) {
+                if (auth.getEnabledTaskTypes().contains(model.getTaskType()) == false) {
+                    logger.warn(
+                        Strings.format(
+                            "The authorization response included the default model: %s, "
+                                + "but did not authorize the assumed task type of the model: %s. Enabling model.",
+                            id,
+                            model.getTaskType()
+                        )
+                    );
+                }
+                enabledConfigIds.add(new DefaultConfigId(model.getInferenceEntityId(), model.getTaskType(), this));
+            }
+        }
+
+        return enabledConfigIds;
+    }
+
+    private Set<String> getEnabledDefaultModelIds(ElasticInferenceServiceAuthorization auth) {
+        var enabledModels = auth.getEnabledModels();
+        var enabledDefaultModelIds = new HashSet<>(defaultModels.keySet());
+        enabledDefaultModelIds.retainAll(enabledModels);
```

Comment:
Return the model ids where there is overlap between what the gateway authorized and the default ones we've defined.